docs/dev-guide/contributing/building.rst

```diff
@@ -8,54 +8,54 @@ Concepts
-pulp-2.4-rhel7, respectively. You can see the full list of Pulp's Koji tags
+called tags. A tag is essentially a grouping of package builds.
+Pulp uses one Koji tag per Pulp X.Y release stream, per distribution, per architecture.
+For example, the 2.6 releases of pulp will build into the pulp-2.6-<distribution> tags in koji.
+You can see the full list of Pulp's Koji tags
 `here <http://koji.katello.org/koji/search?match=glob&type=tag&terms=pulp*>`_.
 
 Another thing to know about Koji is that once a particular NEVRA (Name, Epoch, Version, Release,
 Architecture) is built in Koji, it cannot be built again. However, it can be tagged into multiple
 Koji tags. For example, if ``python-celery-3.1.11-1.el7.x86_64`` is built into the
-``pulp-2.4-beta-rhel7`` tag and you wish to add that exact package in the ``pulp-2.4-rhel7`` tag,
+``pulp-2.4-rhel7`` tag and you wish to add that exact package in the ``pulp-2.5-rhel7`` tag,
 you cannot build it again. Instead, you must tag that package for the new tag. You will see later
 on in this document that Pulp has a tool to help you do this.
 
+Pulp release and testing builds are collections of components that are versioned independently.
+For example, the core Pulp server may be at version 2.6 while pulp_docker may be at version 1.0.
+This assembly is accomplished using release definitions specified in the
+``pulp_packaging/ci/config/releases/<build-name>.yaml`` files. Each file specifies the detail of a
+build that the Pulp build scripts can later can assemble. The components within that
+file specify the target koji tag as well as the individual git repositories and branches that
+will be assembled as part of a build. In addition it specifies the directory within
+https://repos.fedorapeople.org/repos/pulp/pulp/testing/automation/ where the build results
+will be published.
+
+Because there is no way to automatically determine when a particular component needs a new version,
+or what that version should be, the build-infrastructure assumes that whatever version is specified
+in the rpm spec file is the final version that is required.  If a release build
+of that version has already been built in koji then those RPMs will be used.  Ideally, whenever
+a branch is forked, or the first commit is pushed into a branch after a release build, the version
+should be incremented.
+
 Tools used when building
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-Pulp has some wrapper scripts in the ``pulp/rel-eng`` directory to assist with
+Test or release builds (with the exclusion of the signing step) may be performed using
+Jenkins.  There are automated jobs that will run nightly which build repositories that can be used
+for validation.  When those jobs are initiated manually there is a parameter to enable the
+release build process in koji.  If a release build is performed with Jenkins you will still need
+to sign the rpms and push them to the final, non automation, location on fedorapeople.
```

Comment:
The ", non automation," bit reads a little strangely to me. Maybe we could just say:

"and manually push them to the final location on fedorapeople."